metakb/deltas/delta.py

```diff
@@ -1,119 +1,124 @@
-"""A module for computing CIViC deltas."""
+"""A module for computing deltas."""
 from metakb import PROJECT_ROOT
 import json
 import logging
 from jsondiff import diff
 from datetime import date
-import pkg_resources
-from metakb.harvesters import CIViC
+from metakb.harvesters import CIViC, MOAlmanac
 
 logger = logging.getLogger('metakb')
 logger.setLevel(logging.DEBUG)
 
 
-class CIVICDelta:
-    """A class for computing CIViC deltas."""
+class Delta:
+    """A class for computing deltas."""
 
-    def __init__(self, main_json, *args, **kwargs):
-        """Initialize the CIVICDelta class.
+    def __init__(self, main_json, src, *args, **kwargs):
+        """Initialize the Delta class.
 
-        :param str main_json: The path to the main CIViC composite json file.
+        :param str main_json: The path to the main composite json file.
+        :param str src: The source to compute the delta on
         """
+        assert src.lower() in ['civic', 'pmkb', 'moa']
```

Comment:
I wonder if we can codify the source list somewhere in our schema as an enum, and draw from that enum to generate this list and other places such a list might be used? I think it would be a nice, self-documenting way to improve code maintainability and query validation.